Upstream: This *is* from upstream.
Source: Upstream, https://svn.kvirc.de/kvirc/changeset/3187
Reason: Fixes crash, https://svn.kvirc.de/kvirc/ticket/424

Index: /trunk/kvirc/src/kvirc/ui/kvi_statusbarapplet.cpp
===================================================================
--- /trunk/kvirc/src/kvirc/ui/kvi_statusbarapplet.cpp (revision 2803)
+++ /trunk/kvirc/src/kvirc/ui/kvi_statusbarapplet.cpp (revision 3187)
@@ -127,16 +127,8 @@
 	{
 		QPainter p(this);
-	// workaround to fix "Warning:QPainter::setCompositionMode: PorterDuff modes not supported on device on win"
-#if defined(COMPILE_ON_WINDOWS) || defined(COMPILE_ON_MINGW)
-			QPalette pal=palette();
-			QColor col=pal.highlight().color();
-			col.setAlpha(127);
-			p.fillRect(rect(),QBrush(col));
-#else
-			
-		p.setCompositionMode(QPainter::CompositionMode_SourceOut);
-		p.fillRect(rect(),Qt::black);
-		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
-#endif
+		QPalette pal=palette();
+		QColor col=pal.highlight().color();
+		col.setAlpha(127);
+		p.fillRect(rect(),QBrush(col));
 	}
 }
Index: /trunk/kvirc/src/kvirc/ui/kvi_statusbar.h
===================================================================
--- /trunk/kvirc/src/kvirc/ui/kvi_statusbar.h (revision 2659)
+++ /trunk/kvirc/src/kvirc/ui/kvi_statusbar.h (revision 3187)
@@ -314,5 +314,4 @@
 	//virtual void paintEvent(QPaintEvent * e);
 	virtual void mousePressEvent(QMouseEvent * e);
-	virtual void mouseMoveEvent(QMouseEvent * e);
 	virtual void mouseReleaseEvent(QMouseEvent * e);
 	virtual void mouseDoubleClickEvent(QMouseEvent * e);
Index: /trunk/kvirc/src/kvirc/ui/kvi_statusbar.cpp
===================================================================
--- /trunk/kvirc/src/kvirc/ui/kvi_statusbar.cpp (revision 2803)
+++ /trunk/kvirc/src/kvirc/ui/kvi_statusbar.cpp (revision 3187)
@@ -487,56 +487,42 @@
 }
 
-void KviStatusBar::mouseMoveEvent(QMouseEvent * e)
-{
-
-	if(!m_pClickedApplet)
-		return;
-	if(!appletExists(m_pClickedApplet))
-		return;
-
-	KviStatusBarApplet * pApplet = (KviStatusBarApplet *) childAt(e->pos());
-
-	if(pApplet == m_pClickedApplet)
-		return;
-
-	// move!
-	if(!pApplet)
-	{
-		pApplet = m_pAppletList->first();
-		while(pApplet)
-		{
-			if(e->pos().x() < (pApplet->x()+pApplet->width()))
+void KviStatusBar::mouseReleaseEvent(QMouseEvent * e)
+{
+	if(e->button() == Qt::LeftButton)
+	{
+		if(m_pClickedApplet && appletExists(m_pClickedApplet))
+		{
+			KviStatusBarApplet * pApplet = (KviStatusBarApplet *) childAt(e->pos());
+
+			if(pApplet == m_pClickedApplet)
+				return;
+
+			// move!
+			if(!pApplet)
 			{
-				break;
-			} else {
-				pApplet = m_pAppletList->next();
+				pApplet = m_pAppletList->first();
+				while(pApplet)
+				{
+					if(e->pos().x() < (pApplet->x()+pApplet->width()))
+					{
+						break;
+					} else {
+						pApplet = m_pAppletList->next();
+					}
+				}
+				if(!pApplet || pApplet == m_pClickedApplet)
+					return; // no way to move
 			}
-		}
-		if(!pApplet || pApplet == m_pClickedApplet)
-			return; // no way to move
-	}
-
-	printf("inverting %d and %d\n",m_pClickedApplet->index(),pApplet->index());
-
-	//swap indexes
-	int oldIndex = m_pClickedApplet->index();
-	m_pClickedApplet->setIndex(pApplet->index());
-	pApplet->setIndex(oldIndex);
-
-	removeWidget(m_pClickedApplet); //Note: This function does not delete the widget but hides it. To add the widget again, you must call both the addWidget() and show() functions.
-	insertPermanentWidget(m_pClickedApplet->index(), m_pClickedApplet);
-	m_pClickedApplet->show();
-
-	m_pClickedApplet->select(false);
-	g_pApp->restoreOverrideCursor();
-	m_pClickedApplet = 0;
-}
-
-void KviStatusBar::mouseReleaseEvent(QMouseEvent * e)
-{
-	if(e->button() == Qt::LeftButton)
-	{
-		if(m_pClickedApplet && appletExists(m_pClickedApplet))
-		{
+
+			//swap indexes
+			int oldIndex = m_pClickedApplet->index();
+			int newIndex = pApplet->index();
+			m_pClickedApplet->setIndex(newIndex);
+			pApplet->setIndex(oldIndex);
+
+			removeWidget(m_pClickedApplet); //Note: This function does not delete the widget but hides it. To add the widget again, you must call both the addWidget() and show() functions.
+			insertPermanentWidget(newIndex, m_pClickedApplet);
+			m_pClickedApplet->show();
+
 			m_pClickedApplet->select(false);
 			g_pApp->restoreOverrideCursor();
